apps/gro/controllers/confirm.js

```diff
@@ -1,36 +1,83 @@
 'use strict';
 
 const BaseController = require('hof').controllers.confirm;
-const path = require('path');
-const i18n = require('hof').i18n;
-const Model = require('../../common/models/email');
+const _ = require('lodash');
+const config = require('../../../config');
+const EmailService = require('../../common/services/email');
+const fields = require('../fields');
+
+delete require.cache[require.resolve('../steps')];
+const stepConfig = require('../steps');
 
 module.exports = class ConfirmController extends BaseController {
 
-  saveValues(req, res, callback) {
-    const locali18n = i18n({
-      path: path.resolve(
-        __dirname, '../translations/__lng__/__ns__.json'
+  formatStepsForEmail(data) {
+    return _(stepConfig)
+      .reject(step => !step.locals || !step.fields || _.every(
+        step.fields,
+        field => data[field] === undefined)
       )
-    });
-
-    locali18n.on('ready', () => {
-
-      const subject = locali18n.translate('pages.email.information.subject');
-      const data = {
-        values: Object.assign({}, req.sessionModel.toJSON(), {
-          reportDate: (new Date()).toISOString()
-        }),
-        email: req.sessionModel.get('email-text')
-      };
-
-      const model = new Model(data);
+      .groupBy(step => step.locals.section)
+      .map((steps, section) => ({
+          section,
+          fields: _(steps)
+            .pluck('fields')
+            .flatten()
+            .reject(field => fields[field].includeInEmail === false)
+            .map(field => ({
+              field,
+              value: data[field]
+            }))
+            .value()
+        })
+      )
+      .value();
+  }
 
-      model.set('template', 'gro');
-      model.set('subject', subject);
+  // eslint-disable-next-line consistent-return
+  saveValues(req, res, callback) {
+    const data = this.formatStepsForEmail(req.sessionModel.toJSON());
+    if (!data.length) {
+      return callback(new Error('No data to send'));
+    }
+    _.first(data).fields.unshift({
+      field: 'submission-date',
+      value: (new Date()).toISOString()
+    });
+    Promise.all([
+      this.sendEmail(config.email.caseworker, 'caseworker', data, req, res),
+      this.sendEmail(req.sessionModel.get('email-address'), 'customer', data, req, res)
+    ]).then(() => callback()).catch(err => callback(err));
+  }
 
-      model.save(callback);
+  sendEmail(to, recipient, data, req, res) {
+    const subject = req.translate('email.information.subject');
```

Comment:
Obvs, a component-like architecture could avoid these issues. Where coupling a controller, view, fields et al would mean the dev would simply name the component they needed in the step. Dreams... or not?